java/ql/src/Security/CWE/CWE-200/TempDirLocalInformationDisclosure2.ql

```diff
@@ -0,0 +1,72 @@
+/**
+ * @name Temporary Directory Local information disclosure
+ * @description Detect local information disclosure via the java temporary directory
+ * @kind path-problem
+ * @problem.severity warning
+ * @precision very-high
+ * @id java/local-information-disclosure
+ * @tags security
+ *       external/cwe/cwe-200
+ */
+
+import TempDirUtils
+import DataFlow::PathGraph
+
+private class MethodFileSystemFileCreation extends Method {
+  MethodFileSystemFileCreation() {
+    getDeclaringType() instanceof TypeFile and
+    (
+      hasName("mkdir") or
+      hasName("createNewFile")
+    )
+  }
+}
+
+private class MethodFilesSystemFileCreation extends Method {
+  MethodFilesSystemFileCreation() {
+    getDeclaringType().hasQualifiedName("java.nio.file", "Files") and
+    hasName("write")
+  }
+}
+
+private abstract class FileCreationSink extends DataFlow::Node {}
+
+private class FileFileCreationSink extends FileCreationSink {
+  FileFileCreationSink() {
+    exists(MethodAccess ma |
+      ma.getMethod() instanceof MethodFileSystemFileCreation and
+      ma.getQualifier() = this.asExpr()
+    )
```

Comment:
You could write:
```suggestion
    exists(MethodFileSystemFileCreation ma |
      ma.getQualifier() = this.asExpr()
    )
```

Reply:
This is not quite right. `MethodFileSystemFileCreation` is a `Method` not a `MethodAccess`

Reply:
Yea, I don't think I can change this to be simpler at all.